### PR TITLE
Update parent pom and use spotbugs

### PR DESCRIPTION
**Pull request by StefanSpieker**
* update parent pom, stapler, min java and jenkins version and use spotbugs.
* use spotbugs annotations and fixed some NonNull annotations

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -90,26 +96,22 @@
             <scope>provided</scope>
         </dependency>
         <dependency>
-            <!-- Baseline of Stapler in 1.625.3, required to fix upper bounds issue in the core -->
             <groupId>org.kohsuke.stapler</groupId>
             <artifactId>stapler</artifactId>
-            <version>1.237</version>
+            <version>1.260</version>
```
Comment (oleg-nenashev) [inline pom.xml:101]: 2.249.1 should include it in the BOM

**Review by oleg-nenashev — APPROVED** [review]
Looks good to me, thanks
No release ETA

File: pom.xml (hunk 1 above)
Comment (StefanSpieker) [reply pom.xml:101]: Should I change the jenkins version or the stapler version?

**Comment by oleg-nenashev** [comment]
Just keep them aligned,

**Comment by StefanSpieker** [comment]
Is it correct to look at [mvnrepository](https://mvnrepository.com/artifact/org.jenkins-ci.main/jenkins-core/2.204.1) what the correct version is? So if I check `<jenkins.core.version>2.204.1</jenkins.core.version>` stapler is `1.258`

**Review by oleg-nenashev — APPROVED** [review]
Sorry, better late than never

**Pull request merged**